src/test/java/fi/aalto/cs/intellij/notifications/NotificationsUtilityTest.java

```diff
@@ -0,0 +1,20 @@
+package fi.aalto.cs.intellij.notifications;
+
+import fi.aalto.cs.intellij.common.BuildInfo;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class NotificationsUtilityTest {
+
+  @Test
+  public void testBetaVersionWarning() {
+    APlusNotifications.BetaVersionWarning notification =
+        new APlusNotifications.BetaVersionWarning(new BuildInfo.Version(0, 22, 315));
+
+    Assert.assertEquals("0.22.315", notification.getVersion().toString());
+    Assert.assertEquals("A+", notification.getGroupId());
+    Assert.assertEquals("A+ Courses plugin is under development", notification.getTitle());
+    Assert.assertThat(notification.getContent(), CoreMatchers.containsString("0.22.315"));
+  }
+}
```

Comment:
if we try to abstract away to some higher levels, then what is done here is to some extent testing of the IJ framework. I'm more or less sure, that the constructor of `Notification` will work as expected, and changing this crucial implementation on the JB's side is unlikely. However, if that is to happen, we can still stick to the "older" (used initially) version of the SDK, but this is highly hypothetical. Or, if we actually aim at assuring, that the `BetaVersionWarning` is still working, then I'd say one single class testing the actual IJ's `Notification` class would be enough for the whole plugin project.

Reply:
I'd see this test more like a description on how we want the `BetaVersionWarning` to work written in a form of a unit test.

It says that we expect this notification has "A+" as group ID, "A+ Courses plugin is under development" as title, and we expect that the version we give to its constructor is mentioned somewhere in the message. Now, if someone wants to modify or refactor or do whatever to that class, they can freely do so. As long as this test passes, it has at least those properties, we (or perhaps just *I*?) cared about. Instead, if this test fails, the modifier should consider whether the expected behavior of the class has changed (and revise the test code) or did s/he mess something up (and fix their code).

Reply:
fair point, let's do so!